terraform/modules/gro_ingestion_service/lambdas/src/functions/publish-event.function.ts

```diff
@@ -0,0 +1,57 @@
+import {DynamoDBRecord, DynamoDBStreamEvent, Handler} from "aws-lambda"
+import {mapToEventRecord} from "../models/EventRecord"
+import {request} from "https"
+
+const apiUrl = process.env.API_URL
+const options = {
+    hostname: apiUrl,
+    path: "/todos",
+    method: "POST",
+    port: 443,
+    headers: {
+        "Content-Type": "application/json",
+    },
+}
+
+const publishRecord = async (record: DynamoDBRecord) => {
+    if (record.eventName !== "INSERT" || !record.dynamodb?.NewImage) {
+        return
+    }
+
+    const eventRecord = mapToEventRecord(record.dynamodb.NewImage)
+    const event = {
+        eventType: "DEATH_NOTIFICATION",
+        eventTime: eventRecord.EventTime,
+        id: eventRecord.hash
+    }
+
+    return new Promise((resolve, reject) => {
```

Comment:
Mixing async on line 16 with promise/callbacks feels odd to me, is there a more consistent way we can write this, or is this to enable the Promise.all syntax later?

Reply:
so returning this promise is how to make the `https` package async